conan/tools/meson/toolchain.py

```diff
@@ -71,7 +73,9 @@ def __init__(self, conanfile, env=os.environ):
         self._shared = self._conanfile.options.get_safe("shared")
         self._fpic = self._conanfile.options.get_safe("fPIC")
         self.definitions = dict()
-        self._env = env
+        self.preprocessor_definitions = dict()
+        self._env = env.copy()
```

Comment:
I'd prefer to wait until we have the new env management there: https://github.com/conan-io/conan/pull/8534, because partially providing it, then breaking could be probably more annoying that having to wait.